openpype/hosts/houdini/plugins/create/create_bgeo.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""Creator plugin for creating pointcache bgeo files."""
+from openpype.hosts.houdini.api import plugin
+from openpype.pipeline import CreatedInstance
+from openpype.lib import EnumDef
+
+
+class CreateBGEO(plugin.HoudiniCreator):
+    """BGEO pointcache creator."""
+    identifier = "io.openpype.creators.houdini.bgeo"
+    label = "BGEO PointCache"
+    family = "bgeo"
+    icon = "gears"
+
+    def create(self, subset_name, instance_data, pre_create_data):
+        import hou
+
+        instance_data.pop("active", None)
+
+        instance_data.update({"node_type": "geometry"})
+        instance_data["bgeo_type"] = pre_create_data.get("bgeo_type")
+
+        instance = super(CreateBGEO, self).create(
+            subset_name,
+            instance_data,
+            pre_create_data)  # type: CreatedInstance
+
+        instance_node = hou.node(instance.get("instance_node"))
+
+        file_path = "{}{}".format(
+            hou.text.expandString("$HIP/pyblish/"),
+            "{}.$F4.{}".format(
+                subset_name,
+                pre_create_data.get("bgeo_type") or "bgeo.sc")
+        )
+        parms = {
+            "sopoutput": file_path
+        }
+
+        if self.selected_nodes:
+            parms["soppath"] = self.selected_nodes[0].path()
+
+            # try to find output node
+            for child in self.selected_nodes[0].children():
+                if child.type().name() == "output":
+                    parms["soppath"] = child.path()
+                    break
```

Comment:
I'm not so sure about this workflow - it seems a bit too magical almost to only do it if the node itself happens to contain an output node. Is this to avoid the issue that an Object level path gets set instead of SOP path? If so, shouldn't we only ever do this if the node is an object level object.

Also, output nodes tend to have an Output Index set in Houdini, right? Shouldn't we be taking the one with the lowest index?

What if I happen to have selected a SOP node that is an HDA/Subnet which itself contains an Output node - I wouldn't want that output node to be set as path but the actual SOP path I had selected.

Reply:
I was thinking about it a lot. I've decided to go with this rather simple way to help set it up because I think there should be something working out-of-the-box and at the same time it shouldn't limit more advanced uses. There is probably no definitive way to handle all cases that can happen so it should either stop assuming anything at all, or implement very robust way of handling it even though there is a good chance it won't be ever used, or use something not perfect but simple to make you go. Honestly, I have no clear opinion on that especially in Houdini.

Good point with the index. It should probably behave like - use selected if it is SOP node, if not, try to find output with the lowest index available.